tests/runtimes/simple-keyvalue/src/lib.rs

```diff
@@ -0,0 +1,270 @@
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Key {
+    #[serde(rename = "key")]
+    #[serde(with = "serde_bytes")]
+    key: Vec<u8>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct KeyValue {
+    #[serde(rename = "key")]
+    #[serde(with = "serde_bytes")]
+    key: Vec<u8>,
+
+    #[serde(rename = "value")]
+    #[serde(with = "serde_bytes")]
+    value: Vec<u8>,
+}
+
+/// Errors emitted by the keyvalue module.
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid argument")]
+    InvalidArgument,
+}
+
+impl sdk::error::Error for Error {
+    fn module(&self) -> &str {
+        MODULE_NAME
+    }
+
+    fn code(&self) -> u32 {
+        match self {
+            Error::InvalidArgument => 1,
+        }
+    }
+}
+
+impl From<Error> for sdk::error::RuntimeError {
+    fn from(err: Error) -> sdk::error::RuntimeError {
+        sdk::error::RuntimeError::new(err.module(), err.code(), &err.msg())
+    }
+}
+
+/// Events emitted by the keyvalue module (none so far).
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Event {
+    // XXX: What if we don't want to have any events?  The SDK seems to require an implementation anyways :/
+    DummyEvent = 1,
+}
+
+impl sdk::event::Event for Event {
+    fn module(&self) -> &str {
+        MODULE_NAME
+    }
+
+    fn code(&self) -> u32 {
+        match self {
+            Event::DummyEvent { .. } => 1,
+        }
+    }
+
+    fn value(&self) -> cbor::Value {
+        cbor::to_value(self)
+    }
+}
+
+/// Parameters for the keyvalue module (none so far).
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Parameters;
+
+#[derive(Error, Debug)]
+pub enum ParameterValidationError {
+    // XXX: Same as above -- what if we don't have any parameters?
+    #[error("dummy error")]
+    DummyError,
+}
+
+impl sdk::module::Parameters for Parameters {
+    type Error = ParameterValidationError;
+
+    fn validate_basic(&self) -> Result<(), Self::Error> {
+        Ok(())
+    }
+}
+
+/// Genesis state for the keyvalue module.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Genesis {
+    #[serde(rename = "parameters")]
+    pub parameters: Parameters,
+}
+
+/// Simple keyvalue runtime module.
+pub struct Module;
+
+impl sdk::module::Module for Module {
+    const NAME: &'static str = MODULE_NAME;
+    const VERSION: u32 = 1;
+    type Error = Error;
+    type Event = Event;
+    type Parameters = Parameters;
+}
+
+impl sdk::module::AuthHandler for Module {}
+impl sdk::module::BlockHandler for Module {}
+
+impl sdk::module::MethodRegistrationHandler for Module {
+    /// Register all supported methods.
+    fn register_methods(methods: &mut sdk::module::MethodRegistry) {
+        methods.register_callable(sdk::module::CallableMethodInfo {
+            name: "keyvalue.insert",
```

Comment:
The convention is to have capitalized method names, e.g. `keyvalue.Insert` :-)

Reply:
OK, I'll fix that, thanks :)